BangazonAPI/Controllers/PaymentTypeController.cs

```diff
@@ -0,0 +1,235 @@
+
+                            paymentTypes.Add(payment);
+
+                        }
+                    }
+
+                    reader.Close();
+                    return Ok(paymentTypes);
+
+                }
+            }
+        }
+
+        // GET api/<PaymentTypeController>/2
+        [HttpGet("{id}", Name = "GetPaymentType")]
+        public async Task<IActionResult> Get([FromRoute] int id)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"
+                        SELECT Id, AcctNumber AS AccountNumber, Name AS PaymentType, CustomerId FROM PaymentType
+                        WHERE Id = @id";
+                    cmd.Parameters.Add(new SqlParameter("@id", id));
+                    SqlDataReader reader = cmd.ExecuteReader();
+
+                    PaymentType paymentType = null;
+
+                    if (reader.Read())
+                    {
+                        paymentType = new PaymentType
+                        {
+                            Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                            AcctNumber = reader.GetString(reader.GetOrdinal("AccountNumber")),
+                            Name = reader.GetString(reader.GetOrdinal("PaymentType")),
+                            CustomerId = reader.GetInt32(reader.GetOrdinal("CustomerId"))
+                        };
+                    }
+                    reader.Close();
+
+                    return Ok(paymentType);
+                }
+            }
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Post([FromBody] PaymentType payment)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"INSERT INTO PaymentType (AcctNumber, Name, CustomerId)
+                                        OUTPUT INSERTED.Id
+                                        VALUES (@acctNum, @name, @custId)";
+                    cmd.Parameters.Add(new SqlParameter("@acctNum", payment.AcctNumber));
+                    cmd.Parameters.Add(new SqlParameter("@name", payment.Name));
+                    cmd.Parameters.Add(new SqlParameter("@custId", payment.CustomerId));
+
+                    int newId = (int)cmd.ExecuteScalar();
+                    payment.Id = newId;
+                    return CreatedAtRoute("GetPaymentType", new { id = newId }, payment);
+                }
+            }
+        }
+
+        //PUT method needs ID to update and New information to change
+        //204NoContent means we aren't getting anything back
+        //Not Found is a 404 Status
+        //CoffeeExists is a method WE wrote to check to see if the coffee id is there or not - returns boolean
```

Comment:
Using reference code is great! But it's also great to get in the habit of cleaning up comments from your reference code before you merge it into master so it doesn't confuse people 😉